pkgs/collection/lib/src/iterable_extensions.dart

```diff
@@ -914,6 +914,33 @@ extension IterableIterableExtension<T> on Iterable<Iterable<T>> {
       };
 }
 
+/// Extension on iterables of [MapEntry].
+///
+/// An [Iterable<MapEntry>] is obtained using [Map.entries], these extensions
+/// make it easy to work on a [Map] as a list of pairs.
+extension IterableMapEntryExtension<K, V> on Iterable<MapEntry<K, V>> {
+  /// Creates a new lazy [Iterable] with all elements whose [MapEntry.key]
+  /// satisfy the predicate [test].
+  Iterable<MapEntry<K, V>> whereKey(bool Function(K) test) =>
+      where((e) => test(e.key));
+
+  /// Creates a new lazy [Iterable] with all elements whose [MapEntry.value]
+  /// satisfy the predicate [test].
+  Iterable<MapEntry<K, V>> whereValue(bool Function(V) test) =>
+      where((e) => test(e.value));
+
+  /// Create an new lazy [Iterable] with [MapEntry.key] from all elements.
+  Iterable<K> get keys => map((e) => e.key);
+
+  /// Create an new lazy [Iterable] with [MapEntry.value] from all elements.
+  Iterable<V> get values => map((e) => e.value);
+
+  /// Create a [Map] from all elements.
```

Comment:
And
```dart
  /// Creates a new [Map<K, V>] with these entries.
  ///
  /// This is ...
```

Reply:
Weird syntax highlighting convinced me that wouldn't work, but it works fine in dartdoc :rocket: 

![image](https://github.com/user-attachments/assets/a3b53043-c304-4428-b19f-aa3f40f4f75a)

Filed https://github.com/dart-lang/dart-syntax-highlight/issues/77 for the syntax issue.